numcodecs/vlen.pyx

```diff
@@ -240,6 +241,7 @@ class VLenBytes(Codec):
         n_items = values.shape[0]
 
         # setup intermediates
+        normed_values = np.empty(n_items, dtype=object)
```

Comment:
To fix the error mentioned above, we need to create a place to store the normalized values

So we allocate an empty array to start. `object` type as we are storing Python objects in it